src/libraries/System.Security.Cryptography/tests/X509Certificates/ExportTests.cs

```diff
@@ -604,5 +754,124 @@ private static (byte[] Pkcs12, TKey key) CreateSimplePkcs12<TKey>() where TKey :
+            const string Pkcs7Encrypted = "1.2.840.113549.1.7.6";
+
+            Pkcs12Info info = Pkcs12Info.Decode(pkcs12, out int read);
+            Assert.Equal(pkcs12.Length, read);
+            Assert.Equal(Pkcs12IntegrityMode.Password, info.IntegrityMode);
+            Assert.True(info.VerifyMac(password), nameof(info.VerifyMac));
+
+            PfxAsn pfxAsn = PfxAsn.Decode(pkcs12, AsnEncodingRules.BER);
+            MacData macData = Assert.NotNull(pfxAsn.MacData);
+            AssertExtensions.GreaterThanOrEqualTo(macData.MacSalt.Length, 20);
+
+            Assert.Equal(expectedIterations, macData.IterationCount);
+            Assert.Equal(expectedMacHashAlgorithm, HashAlgorithmName.FromOid(macData.Mac.DigestAlgorithm.Algorithm));
+            Assert.Null(macData.Mac.DigestAlgorithm.Parameters);
+
+            Assert.Equal(Pkcs7Data, pfxAsn.AuthSafe.ContentType);
+            byte[] safeContents = AsnDecoder.ReadOctetString(pfxAsn.AuthSafe.Content.Span, AsnEncodingRules.BER, out _);
+
+            AsnValueReader authSafeReader = new AsnValueReader(safeContents, AsnEncodingRules.BER);
+            AsnValueReader sequenceReader = authSafeReader.ReadSequence();
+            authSafeReader.ThrowIfNotEmpty();
+
+            int certs = 0;
+            int keys = 0;
+
+            while (sequenceReader.HasData)
+            {
+                ContentInfoAsn.Decode(ref sequenceReader, safeContents, out ContentInfoAsn contentInfo);
+
+                if (contentInfo.ContentType == Pkcs7Encrypted)
+                {
+                    EncryptedDataAsn encryptedData = EncryptedDataAsn.Decode(contentInfo.Content, AsnEncodingRules.BER);
+                    AlgorithmIdentifierAsn algorithmIdentifier = encryptedData.EncryptedContentInfo.ContentEncryptionAlgorithm;
+                    AssertEncryptionAlgorithm(algorithmIdentifier);
+                }
+            }
+
+            foreach (Pkcs12SafeContents pkcs12SafeContents in info.AuthenticatedSafe)
+            {
+                bool wasEncryptedSafe = false;
+
+                if (pkcs12SafeContents.ConfidentialityMode == Pkcs12ConfidentialityMode.Password)
+                {
+                    wasEncryptedSafe = true;
+                    pkcs12SafeContents.Decrypt(password);
+                }
+
+                foreach (Pkcs12SafeBag safeBag in pkcs12SafeContents.GetBags())
+                {
+                    if (safeBag is Pkcs12ShroudedKeyBag shroudedKeyBag)
+                    {
+                        EncryptedPrivateKeyInfoAsn epki = EncryptedPrivateKeyInfoAsn.Decode(
+                            shroudedKeyBag.EncryptedPkcs8PrivateKey,
+                            AsnEncodingRules.BER);
+                        AssertEncryptionAlgorithm(epki.EncryptionAlgorithm);
+                        keys++;
+                    }
+                    else if (safeBag is Pkcs12CertBag && wasEncryptedSafe)
+                    {
+                        certs++;
+                    }
```

Comment:
This is definitely valid in context... but for a generic PFX it's overcounting certs (not filtering out SDSI certs, or whatever that "other" choice is), and undercounting keys (because they could be unshrouded).

Since it's just measuring that we did what we were asked to do... it's probably exactly what we want.

Reply:
Yeah this was very much intentional. It counts what we care about. It also only counts certs that were in an encrypted safe.